src/test/java/seedu/letsgethired/model/application/InternApplicationComparatorTest.java

```diff
@@ -3,51 +3,134 @@
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNotEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
 import static seedu.letsgethired.logic.parser.CliSyntax.PREFIX_COMPANY;
 import static seedu.letsgethired.logic.parser.CliSyntax.PREFIX_CYCLE;
+import static seedu.letsgethired.logic.parser.CliSyntax.PREFIX_DEADLINE;
+import static seedu.letsgethired.logic.parser.CliSyntax.PREFIX_ROLE;
+import static seedu.letsgethired.logic.parser.CliSyntax.PREFIX_STATUS;
 import static seedu.letsgethired.logic.parser.SortOrder.ASCENDING;
 import static seedu.letsgethired.logic.parser.SortOrder.DESCENDING;
-import static seedu.letsgethired.model.application.InternApplicationComparator.createCompositeComparator;
-import static seedu.letsgethired.model.application.InternApplicationComparator.getComparator;
+import static seedu.letsgethired.testutil.TypicalInternApplications.JANE_STREET;
+import static seedu.letsgethired.testutil.TypicalInternApplications.OPTIVER;
 
 import java.util.List;
 
 import org.junit.jupiter.api.Test;
 
 public class InternApplicationComparatorTest {
+    @Test
+    public void getComparator() {
+        InternApplicationComparator companyAscending = InternApplicationComparator
+                .getComparator(PREFIX_COMPANY, ASCENDING);
+        InternApplicationComparator companyDescending = InternApplicationComparator
+                .getComparator(PREFIX_COMPANY, DESCENDING);
+
+        InternApplicationComparator cycleAscending = InternApplicationComparator
+                .getComparator(PREFIX_CYCLE, ASCENDING);
+        InternApplicationComparator cycleDescending = InternApplicationComparator
+                .getComparator(PREFIX_CYCLE, DESCENDING);
+
+        InternApplicationComparator roleAscending = InternApplicationComparator
+                .getComparator(PREFIX_ROLE, ASCENDING);
+        InternApplicationComparator roleDescending = InternApplicationComparator
+                .getComparator(PREFIX_ROLE, DESCENDING);
+
+        InternApplicationComparator statusAscending = InternApplicationComparator
+                .getComparator(PREFIX_STATUS, ASCENDING);
+        InternApplicationComparator statusDescending = InternApplicationComparator
+                .getComparator(PREFIX_STATUS, DESCENDING);
+
+        assertNotNull(companyAscending);
+        assertNotNull(companyDescending);
+
+        assertNotNull(cycleAscending);
+        assertNotNull(cycleDescending);
+
+        assertNotNull(roleAscending);
+        assertNotNull(roleDescending);
+
+        assertNotNull(statusAscending);
+        assertNotNull(statusDescending);
+
+        InternApplicationComparator deadlineDescending = InternApplicationComparator
```

Comment:
Oh? We have this already? But I thought it wasn't implemented by Zhi Sheng yet hahaha